addons/account/models/account_move.py

```diff
@@ -2643,6 +2643,20 @@ def action_duplicate(self):
         action['res_id'] = self.copy().id
         return action
 
+    def _notify_get_groups(self, msg_vals=None):
+        # OVERRIDE
+        """ Give access button to users and portal customer as portal is integrated
+        in sale. Customer and portal group have probably no right to see
+        the document so they don't have the access button. """
+        groups = super(AccountMove, self)._notify_get_groups(msg_vals=msg_vals)
+
+        self.ensure_one()
+        if self.type != 'entry':
+            for group_name, _group_method, group_data in groups:
+                if group_name not in ('customer', 'portal'):
+                    group_data['has_button_access'] = True
```

Comment:
Isn't this just giving access to all registered users even if they have nothing to do with accounting/sales/etc.
And to all other potential groups?

It seems like too much to me

Reply:
This "button" appears in the e-mail of someone who actively follows an invoice/refund (not an entry), so I guess they have access to it.

![immagine](https://user-images.githubusercontent.com/1665365/157031012-5e62c44d-d4ea-49f7-af20-a25cee010218.png)

The code is adapted from `sale_order.py`
https://github.com/odoo/odoo/blob/2e2004b7cda8ea4dadb8de06991b7a2c91992661/addons/sale/models/sale_order.py#L1030-L1041

I discussed with @tde-banana-odoo  who told me to limit the scope of this change to accounting in stable (especially with v13) and that he would do a more comprehensive bugfix/refactor of the `portal_customer` button_access , see the PR above.

And apparently he did, as in `master` we have `button_access = True` for the `portal_customer` group https://github.com/odoo-dev/odoo/blob/a043037bb78ae87951f9e3414f11c66dca7ed4f0/addons/portal/models/portal_mixin.py#L82